core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -352,7 +352,14 @@ class LogSegment(val log: FileMessageSet,
         // If the log is empty, we return time elapsed since the segment is created.
         return now - created
     }
-    now - {if (rollingBasedTimestamp.get >= 0) rollingBasedTimestamp.get else created}
+    // If the first message has a timestamp we use the message timestamp to determine when to roll a segment.
+    // Otherwise, we roll the log segment based on the create time of the log segment. We do this to avoid
+    // potential frequent log segment rolling during the partition reassignment. The log rolling will only based on
+    // message timestamp after the topic start to use message format 0.10.0 and above.
```

Comment:
It seems like this duplicates the scaladoc of the method a little. Maybe consolidate it there?